File: greatest_common_factor/greatest_common_factor.cpp
#include <iostream>

using namespace std;


/**
 * @brief calculate greatest common factor for a and b
 * @param a int
 * @param b int
 * @return int
 */
int greatest_common_factor(int a, int b) {
    while(b != 0) {
        int reminder = a % b;
        a = b;
        b = reminder;
    }
    return a;
}


int main(){
    cout << greatest_common_factor(8, 16) << endl;
    return 0;
}

File: greatest_common_factor/greatest_common_factor.py
# Description: 
# Author: Kireev Georgiy <metallerok@gmail.com>
# Copyright (C) 2019 by Kireev Georgiy


def greatest_common_factor(a, b):
    """
    calculate greatest common factor for a and b
    :type a: object
    :param a: int
    :param b: int
    :return: int
    """
    while not b == 0:
        reminder = a % b
        a = b
        b = reminder

    return a


if __name__ == '__main__':
    print(greatest_common_factor(8, 16))
